packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -32,6 +32,89 @@ export interface GitPushFilter {
   readonly tagsIncludes?: string[];
 }
 
+/**
+ * Git pull request filter for trigger.
+ */
+export interface GitPullRequestFilter {
+  /**
+   * The list of patterns of Git branches that, when pull request events occurs, are
+   * to be excluded from starting the pipeline.
+   *
+   * You can filter with glob patterns. The `branchesExcludes` takes priority
+   * over the `branchesIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no branches.
+   */
+  readonly branchesExcludes?: string[];
+
+  /**
+   * The list of patterns of Git branches that, when pull request events occurs, are
+   * to be included as criteria that starts the pipeline.
+   *
+   * You can filter with glob patterns. The `branchesExcludes` takes priority
+   * over the `branchesIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no branches.
+   */
+  readonly branchesIncludes?: string[];
+
+  /**
+   * The list of patterns of Git repository file paths that, when pull request events occurs,
+   * are to be excluded from starting the pipeline.
+   *
+   * You can filter with glob patterns. The `filePathsExcludes` takes priority
+   * over the `filePathsIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no filePaths.
+   */
+  readonly filePathsExcludes?: string[];
+
+  /**
+   * The list of patterns of Git repository file paths that, when pull request events occurs,
+   * are to be included as criteria that starts the pipeline.
+   *
+   * You can filter with glob patterns. The `filePathsExcludes` takes priority
+   * over the `filePathsIncludes`.
+   *
+   * Maximum length of this array is 8.
+   *
+   * @default - no filePaths.
+   */
+  readonly filePathsIncludes?: string[];
+
+  /**
+   * The field that specifies which pull request events to filter on (opened, updated, closed)
+   * for the trigger configuration.
+   *
+   * @default - all events.
+   */
+  readonly events?: GitPullRequestEvent[];
```

Comment:
Can you please verify if specifying duplicate events is allowed?
Eg, `[GitPullRequestEvent.OPEN, GitPullRequestEvent.OPEN]`

Reply:
It is allowed but will be deduplicated by `Set` without errors.

https://github.com/go-to-k/aws-cdk/blob/7891faa1ae26b91e01f14b87ab4b8a783f4b0ee5/packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts#L285

The unit test could cover it.

https://github.com/go-to-k/aws-cdk/blob/7891faa1ae26b91e01f14b87ab4b8a783f4b0ee5/packages/aws-cdk-lib/aws-codepipeline/test/triggers.test.ts#L706-L707